core/Db/Adapter.php

```diff
@@ -45,14 +46,20 @@ public static function factory($adapterName, & $dbInfos, $connect = true)
             $infos[$key] = $val;
         }
 
-        $adapter   = new $className($infos);
+        $adapter = new $className($infos);
 
         if ($connect) {
-            $adapter->getConnection();
-
-            Zend_Db_Table::setDefaultAdapter($adapter);
-            // we don't want the connection information to appear in the logs
-            $adapter->resetConfig();
+            try {
+                $adapter->getConnection();
+
+                Zend_Db_Table::setDefaultAdapter($adapter);
+                // we don't want the connection information to appear in the logs
+                $adapter->resetConfig();
+            } catch(\Exception $e) {
+                // we don't want the exception to leak information
+                $msg = self::safeExceptionMessage($e->getMessage());
+                throw new \Exception($msg);
```

Comment:
Would it be better to rethrow the exception if nothing is matched in `safeExceptionMessage()`? This would preserve the original stack trace.

Reply:
Yes good idea, thanks